app/serializers/chat_channel_serializer.rb

```diff
@@ -45,4 +50,52 @@ def chatable
       nil
     end
   end
+
+  def archive
+    object.chat_channel_archive
+  end
+
+  def include_archive_status?
+    scope.is_staff? && archive.present?
+  end
+
+  def include_archive_completed?
+    include_archive_status?
+  end
+
+  def archive_completed
+    archive.complete?
+  end
+
+  def include_archive_failed?
+    include_archive_status?
```

Comment:
In many serializes in core, I see we are using `alias_method`

Reply:
Ah yeah that would be better, forgot about that!